RFCs/0028-BlameCollector-Hang-Detection.md

```diff
@@ -1,44 +1,88 @@
+the `procdump.exe` and `procdump64.exe` tools.
+This path may be `%ProgramData%\chocolatey\bin\` if you use `choco install procdump` to acquire these tools.
 
-```
+Add the required settings as shown below in the .runsettings file sample below:
+
+```xml
 <RunSettings>
-    <DataCollectionRunSettings>
-        <DataCollectors>
-            <DataCollector friendlyName="blame" enabled="True">
-                <Configuration>
-                    <CollectDumpOnTestSessionHang TestTimeout="300000" DumpType="mini" />
-                    <ResultsDirectory>%AGENT_TEMPDIRECTORY%\TestResults</ResultsDirectory>
-                </Configuration>
-            </DataCollector>
-        </DataCollectors>
-    </DataCollectionRunSettings>
+  <DataCollectionRunSettings>
+    <DataCollectors>
+      <DataCollector friendlyName="blame" enabled="True">
+        <Configuration>
+          <CollectDumpOnTestSessionHang TestTimeout="300000" DumpType="mini" />
+          <ResultsDirectory>%AGENT_TEMPDIRECTORY%</ResultsDirectory>
+        </Configuration>
+      </DataCollector>
+    </DataCollectors>
+  </DataCollectionRunSettings>
 </RunSettings>
 ```
 
-Note: This works along with (but can also be used independent of) the existing collect dump option introduced in https://github.com/microsoft/vstest-docs/blob/master/RFCs/0024-Blame-Collector-Options.md
+Note: This works along with (but can also be used independent of) the existing collect dump option introduced in [RFC 0024](0024-Blame-Collector-Options.md).
 
 ## Supported options
-DumpType: If you choose to collect a full process dump. It takes values mini/full. By default, a mini dump will be created.
-TestTimeout: Duration of inactivity in milliseconds (no test events from the test host) after which the data collector assumes a hang has occurred and proceeds to collect a dump and kill the test host process.
-ResultsDirectory: Location to temporarily store the collected process dump
+
+|Option|Description|
+|--|--|
+`DumpType`|If you choose to collect a full process dump. It takes values `mini`/`full`. By default, a mini dump will be created.
+`TestTimeout`|Duration of inactivity in milliseconds (no test events from the test host) after which the data collector assumes a hang has occurred and proceeds to collect a dump and kill the test host process.
+`ResultsDirectory`|The path must exist, but it is not used. Test results go to a `TestResults` folder under the test project directory when `dotnet test` is used locall, or under `$(Agent.TempDirectory)\<guid>` when the `DotNetCoreCLI` task is used on Azure Pipelines.
+
+## Invoking the tests
+
+Add the `--settings path-to\test.runsettings` parameter to `dotnet test` or `vstest.console.exe`.
+Note that this will cause non-Windows agents to fail, so only add this switch on Windows agents.
+
+## Outputs
+
+The `Sequence_guid.xml` and .dmp files will be placed under the test project source directory in a `TestResults\<guid>\` folder.
```

Comment:
This is only created if the blame logger is enabled. Configuring hang dump alone does not create this. This is already referenced in the previous Blame RFC, maybe we can add a line under a section called Recommended Usage to say use this along with the blame logger for best results.

  <Loggers>
      <Logger friendlyName="blame" enabled="True" />
    </Loggers>

Reply:
Isn't the blame logger required to be enabled before turning on the dmp collector? To turn on dumping you need xml elements inside another blame element:

```xml
      <DataCollector friendlyName="blame" enabled="True">
        <Configuration>
          <CollectDump DumpType="full" />
          <CollectDumpOnTestSessionHang TestTimeout="30000" DumpType="full" />
          <!-- This ResultsDirectory must exist, but it is totally ignored. -->
          <ResultsDirectory>%BUILD_ARTIFACTSTAGINGDIRECTORY%</ResultsDirectory>
        </Configuration>
      </DataCollector>
```

What does this look like without the sequence file but with dmp files?

Reply:
Actually this is the section responsible for activating the blame logger (separate entity actually) which generates the sequence file

```
  <LoggerRunSettings>
        <Loggers>	    <Loggers>
            <Logger friendlyName="blame" enabled="True" />	      <Logger friendlyName="blame" enabled="True" />
        </Loggers>	    </Loggers>
    </LoggerRunSettings>	  </LoggerRunSettings> 
```

Reply:
The logger and the blame datacollector can be used independent of each other. Infact they leverage 2 different extensibility points of vstest

Reply:
I observe the Sequence file being generated with or without the `Logger` snippet you just prescribed @ShreyasRmsft. As long as this block is present, the Sequence file is created on crashes:

```xml
  <DataCollectionRunSettings>
    <DataCollectors>
      <DataCollector friendlyName="blame" enabled="True">
        <Configuration>
          <CollectDump DumpType="full" />
          <CollectDumpOnTestSessionHang TestTimeout="30000" DumpType="full" />
          <!-- This ResultsDirectory must exist, but it is totally ignored. -->
          <ResultsDirectory>d:\temp</ResultsDirectory>
        </Configuration>
      </DataCollector>
    </DataCollectors>
  </DataCollectionRunSettings>
```

Reply:
Are you adding the --blame option via the console?

Reply:
No. I only am adding the `-s` switch (and sometimes the `-r` switch).

Reply:
np, going ahead and merging this. I had a few things i wanted to enhance on the vstest usage side, will do it in a separate pr.